libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx

```diff
@@ -0,0 +1,339 @@
+import { useFormContext } from 'react-hook-form'
+import { FC, useEffect, useState } from 'react'
+
+interface VehicleDetails {
+  permno: string
+  make: string
+  color: string
+  isDebtLess?: boolean
+  validationErrorMessages?: VehicleValidationErrorMessage[]
+  requireMilage?: boolean
+}
+
+interface Props extends FieldBaseProps {
+  field: FindVehicleField
+}
+
+function isVehicleOwnerchangeChecksByPermno(
+  response: unknown,
+): response is VehicleOwnerchangeChecksByPermno {
+  return (
+    response !== null &&
+    typeof response === 'object' &&
+    '__typename' in response &&
+    response['__typename'] === 'VehicleOwnerchangeChecksByPermno'
+  )
+}
+
+function isVehiclePlateOrderChecksByPermno(
+  response: unknown,
+): response is VehiclePlateOrderChecksByPermno {
+  return (
+    response !== null &&
+    typeof response === 'object' &&
+    '__typename' in response &&
+    response['__typename'] === 'VehiclePlateOrderChecksByPermno'
+  )
+}
+
+function isVehicleOperatorChangeChecksByPermno(
+  response: unknown,
+): response is VehicleOperatorChangeChecksByPermno {
+  return (
+    response !== null &&
+    typeof response === 'object' &&
+    '__typename' in response &&
+    response['__typename'] === 'VehicleOperatorChangeChecksByPermno'
+  )
+}
+
+function extractVehicleDetails(
+  response:
+    | VehicleOwnerchangeChecksByPermno
+    | VehiclePlateOrderChecksByPermno
+    | VehicleOperatorChangeChecksByPermno,
+): VehicleDetails {
+  // Use type guards to determine the response type and access properties safely
+  if (isVehicleOwnerchangeChecksByPermno(response)) {
+    return {
+      permno: response.basicVehicleInformation?.permno || '',
+      make: response.basicVehicleInformation?.make || '',
+      color: response.basicVehicleInformation?.color || '',
+      isDebtLess: response.isDebtLess ?? true,
+      validationErrorMessages: response.validationErrorMessages ?? [],
+      requireMilage: response.basicVehicleInformation?.requireMileage || false,
+    }
+  } else if (isVehiclePlateOrderChecksByPermno(response)) {
+    return {
+      permno: response.basicVehicleInformation?.permno || '',
+      make: response.basicVehicleInformation?.make || '',
+      color: response.basicVehicleInformation?.color || '',
+      requireMilage: response.basicVehicleInformation?.requireMileage || false,
+    }
+  } else if (isVehicleOperatorChangeChecksByPermno(response)) {
+    return {
+      permno: response.basicVehicleInformation?.permno || '',
+      make: response.basicVehicleInformation?.make || '',
+      color: response.basicVehicleInformation?.color || '',
+      isDebtLess: response.isDebtLess ?? true,
+      validationErrorMessages: response.validationErrorMessages ?? [],
+      requireMilage: response.basicVehicleInformation?.requireMileage || false,
+    }
+  } else {
+    // Handle unexpected response types
+    throw new Error('Unexpected response type')
+  }
+}
```

Comment:
The function repeats the same code structure for different types. This could be simplified by first checking for the existence of basicVehicleInformation and then extracting the needed properties.